ravenframework/PluginBaseClasses/CodePluginBase.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on October 28, 2022
+
+@author: wangc
+"""
+#External Modules---------------------------------------------------------------
+import abc
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PluginBase import PluginBase
+from ..CodeInterfaceClasses import CodeInterfaceBase
+from ..CodeInterfaceClasses import factory
+#Internal Modules End-----------------------------------------------------------
+
+class CodePluginBase(PluginBase, CodeInterfaceBase):
+  """
+    This class represents a specialized class from which each Code interface plugins must inherit from
+  """
+  # List containing the methods that need to be checked in order to assess the
+  # validity of a certain plugin.
+  _methodsToCheck = ['generateCommand', 'createNewInput']
+  entityType = 'Code'
+  _interfaceFactory = factory
+  ##################################################
+  # Plugin APIs
+  ##################################################
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+
+  def initialize(self, runInfo, oriInputFiles):
+    """
+      Method to initialize the run of a new step
+      @ In, runInfo, dict,  dictionary of the info in the <RunInfo> XML block
+      @ In, oriInputFiles, list, list of the original input files
+      @ Out, None
+    """
+    super().initialize(runInfo, oriInputFiles)
+
+  ##############################
+  #       Required Methods
+  ##############################
+  def _readMoreXML(self, xmlNode):
```

Comment:
It might be worth adding the ability for a CodeInterface to support using InputData since we are creating a new base class.

Reply:
Theoretically we can add this capability easily since the code interface base class inherits from the BaseInterface class. We can add getInputSpecifications and handleInput in the CodePluginBase. However, there are several issues that prevents me to expose these methods to the plugin developers:
1. All code interfaces are using readMoreXML, we either need to change all of them to use InputData, or allow both readMoreXML and InputData to be used. I would like to change all the code interfaces to use InputData, but the work is out of the scope, and will have significant impacts on the code interface developers. In addition, we need to have a Team discussion if we want to switch to use InputData.
2. We are still using readMoreXML and InputData in our base classes (i.e., Assembler.py and BaseInterface.py), which make our base classes  a little bit messy. I would like to find some resources to fix this issues before we allow InputData to be used in Code Interfaces.
3. If we allow the InputData to be used, and I think we are going in that direction. We need to have a complete documentation for plugin developers about How to use InputData. Currently, we do not have it. It can be a burden for the plugin developer.

Reply:
Theoretically we can add this capability easily since the code interface base class inherits from the BaseInterface class. We can add getInputSpecifications and handleInput in the CodePluginBase. However, there are several issues that prevents me to expose these methods to the plugin developers:
1. All code interfaces are using readMoreXML, we either need to change all of them to use InputData, or allow both readMoreXML and InputData to be used. I would like to change all the code interfaces to use InputData, but the work is out of the scope, and will have significant impacts on the code interface developers. In addition, we need to have a Team discussion if we want to switch to use InputData.
2. We are still using readMoreXML and InputData in our base classes (i.e., Assembler.py and BaseInterface.py), which make our base classes  a little bit messy. I would like to find some resources to fix this issues before we allow InputData to be used in Code Interfaces.
3. If we allow the InputData to be used, and I think we are going in that direction. We need to have a complete documentation for plugin developers about How to use InputData. Currently, we do not have it. It can be a burden for the plugin developer.

Reply:
If there is a way to allow InputData to be optionally used, that would be good, but I will merge this without that change if needed.

Reply:
I have added the getInputSpecifications and _handleInput method in the CodeInterfaceBase class, and I just feel it is not ready to provide this method to CodePlugins.

Reply:
Okay, thanks.